### PR TITLE
amp_kart_bringup: changed server plugins

### PR DESCRIPTION
**Pull request by jay-jhk-lee**
## Brief Description

controller server: default -> regulated pure pursuit
planner server: default -> smac planner

## Details

Increased cost_penalty parameter from default because the kart would collide with obstacles and become stuck; higher cost_penalty results in wider turns around obstacles, but removes that issue.

Fixed issue of paths not being straight when navigating

### CODE REVIEW

File: src/amp_kart_bringup/params/nav2.params.yaml [hunk 1]
```diff
@@ -87,77 +87,49 @@ bt_navigator_rclcpp_node:
...
 controller_server:
   ros__parameters:
-    use_sim_time: True
     controller_frequency: 20.0
     min_x_velocity_threshold: 0.001
     min_y_velocity_threshold: 0.5
     min_theta_velocity_threshold: 0.001
     progress_checker_plugin: "progress_checker"
-    goal_checker_plugin: "goal_checker"
+    goal_checker_plugins: "goal_checker"
```
Comment (alanssitis) [inline src/amp_kart_bringup/params/nav2.params.yaml:95]:
```suggestion
    goal_checker_plugin: "goal_checker"
```
shouldn't be changed

File: src/amp_kart_bringup/params/nav2.params.yaml [hunk 2]
```diff
@@ -87,77 +87,49 @@ bt_navigator_rclcpp_node:
 
 controller_server:
   ros__parameters:
-    use_sim_time: True
```
Comment (alanssitis) [inline src/amp_kart_bringup/params/nav2.params.yaml:90]: shouldn't be removed imo

File: src/amp_kart_bringup/params/nav2.params.yaml [hunk 3]
```diff
@@ -261,15 +233,27 @@ map_saver:
 
 planner_server:
   ros__parameters:
-    expected_planner_frequency: 20.0
```
Comment (alanssitis) [inline src/amp_kart_bringup/params/nav2.params.yaml:264]: what happens when you add this back?

**Review by alanssitis — CHANGES_REQUESTED** [review]
@jay-jhk-lee look at the comments I made, just minor changes or clarifying why a change was made.

File: src/amp_kart_bringup/params/nav2.params.yaml (hunk 3 above)
Comment (jay-jhk-lee) [reply src/amp_kart_bringup/params/nav2.params.yaml:264]: expected_planner_frequency is a parameter
that sets the lower bound of current current planner frequency
and sends warnings if current goes below expected.
doesn't change the actual functionality of the planner

**Review by alanssitis — APPROVED** [review]
I trust, lgtm

**Pull request merged**